packages/compass-explain-plan/src/components/explain-json/explain-json.tsx

```diff
@@ -0,0 +1,39 @@
+import type { FunctionComponent } from 'react';
+import { useCallback, useMemo } from 'react';
+import React from 'react';
+import HadronDocument from 'hadron-document';
+import { Document } from '@mongodb-js/compass-crud';
+import { css, KeylineCard, spacing } from '@mongodb-js/compass-components';
+
+const cardStyles = css({ padding: spacing[2] });
+interface ExplainJSONProps {
+  originalExplainData: Record<string, unknown>;
+}
+
+const ExplainJSON: FunctionComponent<ExplainJSONProps> = ({
+  originalExplainData,
+}) => {
+  const copyToClipboard = useCallback(() => {
+    void navigator.clipboard.writeText(JSON.stringify(originalExplainData));
+  }, [originalExplainData]);
+
+  const doc = useMemo(
+    () => new HadronDocument(originalExplainData),
+    [originalExplainData]
+  );
```

Comment:
Document component will do the memo for you, you don't need to do it manually here